pkg/labels/selector.go

```diff
@@ -47,6 +47,18 @@ func Everything() Selector {
 	return LabelSelector{}
 }
 
+type nothingSelector struct{}
+
+func (n nothingSelector) Matches(_ Labels) bool                         { return false }
+func (n nothingSelector) Empty() bool                                   { return false }
+func (n nothingSelector) String() string                                { return "<null>" }
+func (n nothingSelector) Add(_ string, _ Operator, _ []string) Selector { return n }
+
+// Nothing returns a selector that matches no labels
+func Nothing() Selector {
```

Comment:
@mikedanese 
Can you help clarify what was the reason for `Nothing()`? If it wants to select nothing, why does it even bother to select?

/cc @xiang90